### DISCUSSION TITLE
[Help]: Carousel resets to first slide during route transition

### DISCUSSION THREAD

**Discussion by michaelprys** [davidjerleke/embla-carousel#962] [Help] [answered]
### Summary

I’m working on a Nuxt 3 application where I use Vue transitions. The problem is that when I navigate between routes and the transition occurs, the carousel resets to the first slide before the transition completes, which looks weird.

I want to persist the carousel on the current slide during route transition (or maintain its state after a page refresh). Is it possible to do with Embla or I need a state management like Pinia? Thank you.

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [X] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
Hi @michaelprys,

Your problem could be related to this maybe?

- #869

If yes, a bug report was created in the `vuejs/core` repository because all other reactive libraries behave as expected but Vue doesn’t. See:

- https://github.com/vuejs/core/issues/10927

Don’t know if the Vue team will do anything about it though because they’re claiming it’s by design.

**Reply by michaelprys** [reply]
I've experienced a similar issue with Splide Vue. Looks like it's a common issue. Hope Vue team fixes it. Thank you for clarifying.
